src/controller/CHIPDeviceController.cpp

```diff
@@ -212,7 +212,10 @@ CHIP_ERROR DeviceController::InitControllerNOCChain(const ControllerInitParams &
     ReturnErrorOnFailure(ConvertX509CertToChipCert(params.controllerNOC, nocSpan));
     ReturnErrorOnFailure(ExtractNodeIdFabricIdFromOpCert(nocSpan, &nodeId, &fabricId));
 
-    auto * fabricTable            = params.systemState->Fabrics();
+    auto * systemState = params.systemState ? params.systemState : mSystemState;
```

Comment:
Change should result in some form of documentation update.

After that it seems ok as it seems to 100% preserve behaviour for existing clients.

Reply:
@andy31415 you mean to add some notion about reusing `mSystemState` to [InitControllerNOCChain description](https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CHIPDeviceController.h#L380-L390)?

Reply:
@andy31415 Doc for the `InitControllerNOCChain` already comprehensive: https://github.com/project-chip/connectedhomeip/blob/master/src/controller/CHIPDeviceController.h#L385-L390

> if you know exactly how this will interact with controller state

More information will just duplicate the code